src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2931,4 +2932,33 @@ BOOST_AUTO_TEST_CASE(ConvolutionLayerPaddingTest)
   module2.Backward(std::move(input), std::move(output), std::move(delta));
 }
 
+/**
+ * Simple Separable Convolution layer test.
+ */
+BOOST_AUTO_TEST_CASE(SimpleSeparableConvolutionLayerTest)
+{
+  arma::mat output, input, delta;
+  Convolution<> module1(1, 4, 2, 2, 1, 1, 0, 0, 3, 3, "valid");
+
+  // Test the forward function.
+  input = arma::linspace<arma::colvec>(0, 8, 9);
+  input.zeros();
+  input(0) = 1.0;
+  input(3) = 3.0;
+  input(4) = 4.0;
+  input(5) = 5.0;
+  input(7) = 1.0;
+  module1.Parameters() = arma::mat(20, 1, arma::fill::zeros);
+  module1.Parameters()(1) = 1.0;
+  module1.Parameters()(2) = 1.0;
+  module1.Parameters()(5) = 2.0;
+  module1.Parameters()(7) = 2.0;
+  module1.Parameters()(8) = 3.0;
+  module1.Parameters()(14) = 4.0;
+  module1.Reset();
+  module1.Forward(std::move(input), std::move(output));
+
+  BOOST_REQUIRE_EQUAL(arma::accu(output), 111);
```

Comment:
Hi @zoq, I needed your help while implementing tests for this layer.
I ran the following test in python and got sum as 111 but when I run the same test using Convolution Layer I get 113. Could you help me in figuring out what I did wrong.
Thanks.